iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/visitors/ProjectContextTest.java

```diff
@@ -19,8 +19,110 @@
+    assertThat(ctx.getProjectResources("namespace1", toPathString("path1/something.yaml"), LimitRange.class)).containsExactly(resource12);
+    assertThat(ctx.getProjectResources("namespace1", toPathString("path1/something.yaml"), TestResource.class)).isEmpty();
+    assertThat(ctx.getProjectResources("namespace2", toPathString("path2/something.yaml"), ServiceAccount.class)).containsExactlyInAnyOrder(resource21, resource22);
+    assertThat(ctx.getProjectResources("namespace1", toPathString("wrong-path/something.yaml"), ServiceAccount.class)).isEmpty();
+    assertThat(ctx.getProjectResources("wrong-namespace", toPathString("random-path/something.yaml"), ServiceAccount.class)).isEmpty();
+  }
+
+  @Test
+  void shouldProvideAccessOnlyToDescendantDirectories() {
+    var builder = ProjectContext.builder(sensorContext.fileSystem());
+
+    var resource1 = mock(TestResource.class);
+    builder.addResource("default", toUri("path1/resource.yaml").toString(), resource1);
+    var resource2 = mock(TestResource.class);
+    builder.addResource("default", toUri("path2/resource.yaml").toString(), resource2);
+    var resource3 = mock(TestResource.class);
+    builder.addResource("default", toUri("path1/subdir1/resource.yaml").toString(), resource3);
+    var resource4 = mock(TestResource.class);
+    builder.addResource("default", toUri("path1/subdir2/resource.yaml").toString(), resource4);
+    var ctx = builder.build();
+
+    assertThat(ctx.getProjectResources("default", toPathString("path1/something.yaml"), TestResource.class)).containsExactlyInAnyOrder(resource1, resource3, resource4);
+    assertThat(ctx.getProjectResources("default", toPathString("path1/subdir1/something.yaml"), TestResource.class)).containsExactlyInAnyOrder(resource3);
+    assertThat(ctx.getProjectResources("default", toPathString("path1/subdir2/something.yaml"), TestResource.class)).containsExactlyInAnyOrder(resource4);
+    assertThat(ctx.getProjectResources("default", toPathString("path1/subdir3/something.yaml"), TestResource.class)).isEmpty();
+    assertThat(ctx.getProjectResources("default", toPathString("path2/something.yaml"), TestResource.class)).containsExactlyInAnyOrder(resource2);
+  }
+
+  @Test
+  void shouldProvideAccessOnlyToResourcesInTheSameHelmChart() {
+    var builder = ProjectContext.builder(sensorContext.fileSystem());
+
+    var resource1 = mock(TestResource.class);
+    builder.addResource("default", toUri("path1/templates/resource.yaml").toString(), resource1);
+    var resource2 = mock(TestResource.class);
+    builder.addResource("default", toUri("path1/Chart.yaml").toString(), resource2);
+    var resource3 = mock(TestResource.class);
+    builder.addResource("default", toUri("path2/resource.yaml").toString(), resource3);
+    var ctx = builder.build();
+
+    try (var ignored = Mockito.mockStatic(HelmFileSystem.class)) {
```

Comment:
In theory you could also build the input files and put them into the filesystem so you don't have to mock them.
But that would go more in the direction of an integration test, so I'm fine with statically mocking here.